### PR TITLE
pre-transaction-actions plugin implementation

### PR DESCRIPTION
**Pull request by mskrgulja**
This is DNF implementation of pre-transactions-actions (originally for YUM.)
For creating it I used post-transaction-actions plugin as a starting point, so in essence this is very similar to post-transaction-actions, but with different hook and action names.

### CODE REVIEW

**Comment by mskrgulja** [comment]
Any news here :-)

File: doc/pre-transaction-actions.rst [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+``package_filter:transaction_state:command``.
+
+``package_filter``
+   A (glob-like) filtering rule applied on the package NEVRA (also in the shortened forms) or
+   package files.
+
+``transaction_state``
+   Filters packages according to their state in the transaction.
+
+   * ``install`` - packages that will be instaled on the system (install)
```
Comment (jrohel) [inline doc/pre-transaction-actions.rst:59]: As I wrote in the review of the source code. Replace "install", "update" and "remove" with "in" and "out".

Based on the description of the "post-transaction-actions" plugin something like this:
   * ``in`` - packages that will appear on the system (downgrade, install, obsolete, reinstall, upgrade)
   * ``out`` - packages that will disappear from the system (downgraded, obsoleted, remove, upgraded)

**Review by jrohel — CHANGES_REQUESTED** [review]
Thank you for your PR.

Please:
- Replace "install", "update" and "remove" with "in" and "out".
- Merge "Corrected PEP 8 issues", "Fixed more PEP 8 issues", "Resolving build issues" commits with the first one. So only two commits remain - "Added the pre-transaction-actions plugin" and "Added myself to Authors file"
- Fix your "git" settings (or create commits under the correct user account). Make your commits contain the github name "mskrgulja" and not "root".
- Remove execute "x" right on "pre-transaction-actions.py" file.

**Comment by jrohel** [comment]
@mskrgulja Thank you for the changes in the code.
I guess I confused you with the term "merge". I wanted something like git "fixup". To clear the history and leave only 2 commits  - "Added the pre-transaction-actions plugin" and "Added myself to Authors file"

**Comment by mskrgulja** [comment]
> @mskrgulja Thank you for the changes in the code. I guess I confused you with the term "merge". I wanted something like git "fixup". To clear the history and leave only 2 commits - "Added the pre-transaction-actions plugin" and "Added myself to Authors file"

You are welcome.
Yes, I was little bit confused (my first time working with Github), but thanks for the clarification.
I will get some time in the next days to solve this.

Take care

**Comment by mskrgulja** [comment]
I don't know how to "fixup" all the commits so I'm closing this pull request and I will create another

**Pull request closed without merging**